### PR TITLE
Update themed components to respect theme from `SpaceKitProvider`

### PR DESCRIPTION
**Pull request by daniman**
This updates all our components which already have theme awareness built in to look for their theme from `SpaceKitProvider` in addition to accepting a `theme` prop. This should be a no-op to all current users (we continue to default to the "light" theme), but should allow users interested in dark components to use the `SpaceKitProvider` to theme all components at once.

Please hide whitespace changes to review.

<details>
  <summary>📦 Published PR as canary version: <code>8.22.1-canary.352.9465.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @apollo/space-kit@8.22.1-canary.352.9465.0
  # or 
  yarn add @apollo/space-kit@8.22.1-canary.352.9465.0
  ```
</details>

### CODE REVIEW

**Review by cheapsteak — APPROVED** [review]
🌔

**Comment by Jephuff** [comment]
I might consider making this a major, it cooould cause some weird behaviour if someone was using the theme provider (no worries if I missed the merge :P )
It would also give us a chance to see if it creates the major version branch when auto deploying :D

File: src/SpaceKitProvider/index.tsx [hunk 1]
```diff
@@ -47,12 +47,25 @@ const SpaceKitSetContext = React.createContext<
  */
 export const SpaceKitProvider: React.FC<Partial<State>> = ({
```
Comment (Jephuff) [inline src/SpaceKitProvider/index.tsx:48]: Ooh could we pick the values, so no one can pass the singleton prop by accident :)
```suggestion
export const SpaceKitProvider: React.FC<Partial<Pick<State, "theme" | "disableAnimations">>> = ({
```

File: src/SpaceKitProvider/index.tsx [hunk 2]
```diff
@@ -47,12 +47,25 @@ const SpaceKitSetContext = React.createContext<
...
+      singletonComponents: {},
+    }),
+    [theme, disableAnimations],
+  );
+  const [state, setState] = React.useState<State>(nextState);
+
+  useEffect(() => {
+    setState((currState) => ({
+      ...nextState,
+      singletonComponents: currState?.singletonComponents,
```
Comment (Jephuff) [inline src/SpaceKitProvider/index.tsx:66]: NBD we don't need the ? here :)
```suggestion
      singletonComponents: currState.singletonComponents,
```

**Review by Jephuff — APPROVED** [review]

File: src/SpaceKitProvider/index.tsx (hunk 1 above)
Comment (daniman) [reply src/SpaceKitProvider/index.tsx:48]: oh right! 🙏🏽

**Comment by apollo-bot2** [comment]
:rocket: PR was released in [`v9.0.0`](https://github.com/apollographql/space-kit/releases/tag/v9.0.0) :rocket:

**Pull request merged**